### PR TITLE
Fix local transport XA Transactions

### PR DESCRIPTION
**Pull request by madhawa-gunasekara**

### CODE REVIEW

File: modules/commons/src/main/java/org/apache/synapse/commons/transaction/TranscationManger.java [hunk 1]
```diff
@@ -388,12 +388,7 @@ public static Transaction getTransaction() throws Exception {
 		    if (log.isDebugEnabled()) {
 			    log.debug("getTransaction Called");
 		    }
-
-		    TransactionManager txMgr = txManagers.get().get(key);
-		    txMgr.begin();
-		    Transaction tx = txMgr.getTransaction();
-		    transactions.get().put(key, tx);
-		    return tx;
+		    return transactions.get().get(key);
```
Comment (djkevincr) [inline modules/commons/src/main/java/org/apache/synapse/commons/transaction/TranscationManger.java]: What if transactions.get().get(key) is null?

File: modules/commons/src/main/java/org/apache/synapse/commons/transaction/TranscationManger.java [hunk 2]
```diff
@@ -388,12 +388,7 @@ public static Transaction getTransaction() throws Exception {
 		    if (log.isDebugEnabled()) {
 			    log.debug("getTransaction Called");
 		    }
-
-		    TransactionManager txMgr = txManagers.get().get(key);
-		    txMgr.begin();
```
Comment (djkevincr) [inline modules/commons/src/main/java/org/apache/synapse/commons/transaction/TranscationManger.java]: Have you run all the tests? There may be invocations getTransaction() which requires txMgr.begin(). How do you handle such cases?

**Comment by djkevincr** [comment]
I am approving this since, we got this reviewed and approved at support PR.

**Comment by madhawa-gunasekara** [comment]
Resolves https://github.com/wso2/product-ei/issues/683

**Pull request merged**